app/src/fdroid/java/org/fossasia/openevent/general/search/location/LocationServiceImpl.kt

```diff
@@ -61,4 +61,47 @@ class LocationServiceImpl(
             }
         }
     }
+
+    @SuppressLint("MissingPermission")
+    override fun getCityName(): Single<String> {
+        val locationManager = context.getSystemService(Context.LOCATION_SERVICE) as? LocationManager
+            ?: return Single.error(IllegalStateException())
+        val geoCoder = Geocoder(context, Locale.getDefault())
+
+        return Single.create { emitter ->
```

Comment:
Add single cancel handler

Reply:
?

Reply:
This will create a memory leak when Single is cancelled. Add logic to handle the cancellation

Reply:
okay